### PR TITLE
Running the Holdings.add and Holdings.subtract method synchronously

### PR DESCRIPTION
**Pull request by Hugo5000**
Running the add and subtract method synchronously to prevent the overriding of balances, for example:

balance = 100

add(10); => balance is 100, adding 10, new balance is 110, schedule BalanceUpdateEvent for next tick;
add(20); => balance is still 100 since its still the same tick, adding 20, new balance is 120, schedule BalanceUpdateEvent for next tick;

-- next tick --

setting balance to 110
setting balance to 120 => first add is ignored"

### CODE REVIEW

File: src/main/java/com/iConomy/system/Holdings.java [hunk 1]
```diff
@@ -116,17 +118,35 @@ public void set(double balance) {
...
+        if (Bukkit.isPrimaryThread()) {
+            double balance = get();
+            double ending = balance - amount;
+    
+            math(amount, balance, ending);
+        }else{
+            new BukkitRunnable(){
+                @Override
+                public void run(){
+                    add(amount);
```
Comment (LlmDl) [inline src/main/java/com/iConomy/system/Holdings.java]: Looks to be a typo/copy-paste mix-up but this should be subtract(amount), no?

**Review by LlmDl — CHANGES_REQUESTED** [review]

File: src/main/java/com/iConomy/system/Holdings.java (hunk 1 above)
Comment (Hugo5000) [reply src/main/java/com/iConomy/system/Holdings.java]: oh yes!

**Comment by Hugo5000** [comment]
I still have one concern with this pull request, that the logging shows the old balance since that isn't inside the sync task.

**Comment by LlmDl** [comment]
I'm looking at the logging side of things and it would appear that it is being done via this sort of code:
`iConomy.getTransactions().insert("[Vault]", playerName, 0.0D, balance, 0.0D, amount, 0.0D);`
This should probably get moved downstream into the Holdings class.

**Comment by LlmDl** [comment]
> I'm looking at the logging side of things and it would appear that it is being done via this sort of code:
> `iConomy.getTransactions().insert("[Vault]", playerName, 0.0D, balance, 0.0D, amount, 0.0D);`
> This should probably get moved downstream into the Holdings class.

I've looked this over and I don't think it will be manageable. The logging is already fairly spotty and I don't really care if its sacrificed over fixing this issue.

The account update events are cancel-able in the end and the log doesn't take that into account either.

**Pull request merged**